lib/new_relic/transaction/complete.ex

```diff
@@ -170,6 +159,40 @@ defmodule NewRelic.Transaction.Complete do
     {[segment_tree], tx_attrs, tx_error, span_events, apdex, tx_metrics}
   end
 
+  defp transform_root_process_segment(tx_attrs) do
+    transformed =
+      tx_attrs
+      |> Map.take([:name, :pid, :start_time, :end_time])
+      |> transform_trace_time_attrs(tx_attrs.start_time)
+      |> transform_trace_name_attrs()
+
+    struct(Transaction.Trace.Segment, transformed)
+  end
+
+  defp transform_function_segments(function_segments, start_time) do
+    for segment <- function_segments do
+      segment = transform_function_segment(segment, start_time)
+      struct(Transaction.Trace.Segment, segment)
+    end
+  end
+
+  defp transform_function_segment(segment, start_time) do
+    segment
+    |> transform_time_attrs()
+    |> transform_trace_time_attrs(start_time)
+    |> transform_trace_name_attrs()
+  end
+
+  defp transform_process_segments(process_segments, start_time) do
+    for segment <- process_segments,
+        time_attr_segment = transform_trace_time_attrs(segment, start_time),
+        time_attr_segment.relative_start_time != time_attr_segment.relative_end_time do
+      segment = transform_trace_name_attrs(time_attr_segment)
+
+      struct(Transaction.Trace.Segment, segment)
+    end
+  end
```

Comment:
I'd prefer for all these functions to have a matching structure so that they can be understood & compared easily.

Also, let's pull out the `Enum.reject` part back to where it was, since it's not transforming the segments, but filtering them.

```suggestion
  defp transform_process_segments(process_segments, start_time) do
    for segment <- process_segments do
      segment =
        segment
        |> transform_trace_time_attrs(start_time)
        |> transform_trace_name_attrs(time_attr_segment)

      struct(Transaction.Trace.Segment, segment)
    end
  end
```

Reply:
The reject logic here is included in the comprehension, which is more performant than another Enum iteration. If the function naming is the issue, I can change it to filter_map_process_segments or something.